moto/ec2/utils.py

```diff
@@ -753,14 +754,19 @@ def public_key_parse(
     return public_key
 
 
-def public_key_fingerprint(public_key: Union[RSAPublicKey, Ed25519PublicKey]) -> str:
-    # TODO: Use different fingerprint calculation methods based on key type and source
-    # see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/verify-keys.html#how-ec2-key-fingerprints-are-calculated
+def public_key_fingerprint(
+    public_key: Union[RSAPublicKey, Ed25519PublicKey], is_created: bool = False
```

Comment:
The naming of `is_created` is a bit confusing for people without any context. Any objections to rename this to something like `use_sha1: bool`?

A comment in the caller function (`random_rsa_key_pair`) can then explain why we're using sha1:

```
# This fingerprint should use sha1 because it's created by EC2
# See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/verify-keys.html#how-ec2-key-fingerprints-are-calculated
```

Reply:
Thanks @bblommers , I will make changes but wait for @viren-nadkarni review first. No point to trigger CI because of these small fixes

Reply:
Alternatively you could consider moving the logic for determining the hash function to a separate function, and pass the hash constructor callable as an argument to this function.

Reply:
Thanks @viren-nadkarni . I think your approach will make the logic more complex. If I understood you correctly, then `public_key_fingerprint` will have 3 args - `public_key, hash_constructor and use_sha1`. We need to pass `use_sha1`, because there is no other way to know if key is newly generated. So I would stick to current solution with `use_sha1` (ex `is_created`) arg

Reply:
No, my thinking was something like this:

```py
def select_hash_algorithm(
        key: RSAPublicKey| Ed25519PublicKey, is_imported: bool
    ) -> SHA1Type | MD5Type | SHA256Type:
    # This returns the hash constructor based on SSH key format and whether key is imported or not
    ...

hash_constructor = select_hash_algorithm(...)

fingerprint: str = public_key_fingerprint(key, hash_constructor)
```

This ensures that functions are modular, decoupled and unit-testable.

Reply:
My bad, got it. Then it is definitely a better solution. Will use it asap